docs/src/main/sphinx/security/file-system-access-control.rst

```diff
@@ -331,6 +359,94 @@ The example below defines the following table access policy:
       ]
     }
 
+.. _system-file-function-rules:
+
+Function rules
+^^^^^^^^^^^^^^
+
+These rules control the user's ability to execute SQL functions, including PTFs.
+
+Each function rule is composed of the following fields:
+
+* ``user`` (optional): ``regex`` to match against user name. Defaults to ``.*``.
+* ``role`` (optional): ``regex`` to match against role names. Defaults to ``.*``.
+* ``group`` (optional): ``regex`` to match against group names. Defaults to ``.*``.
+* ``catalog`` (optional): ``regex`` to match against catalog name. Defaults to
+  ``.*``.
+* ``schema`` (optional): ``regex`` to match against schema name. Defaults to ``.*``.
+* ``function`` (optional): ``regex`` to match against function names. Defaults to
+  ``.*``.
+* ``privileges`` (required): zero or more of ``EXECUTE``, ``GRANT_EXECUTE``.
+* ``function_kinds`` (required): one or more of ``AGGREGATE``, ``SCALAR``,
+  ``TABLES``, ``WINDOW``. When a user defines a rule for ``AGGREGATE``, ``SCALAR``
+  or ``WINDOW`` functions, the ``catalog`` and ``schema`` fields are disallowed.
+
+
+To restore previous behavior, which denied all PTFs from any catalog, use the following
```

Comment:
What is "previous"? I would prefer to write to achieve X behavior, do y configuration, where you need to explain what X is.

Reply:
Reword this to say what the setup does.. and ignore the fact that this was the default in older releases.

Reply:
Reworded